### PR TITLE
Replace stackdriver logs with pod logs using kubectl

### PR DESCRIPTION
**Pull request by chxchx**
**Release note**:

```release-note
None
```

### CODE REVIEW

File: tests/e2e/util/commonUtils.go [hunk 1]
```diff
@@ -68,16 +68,27 @@ func WriteTempfile(tmpDir, prefix, suffix, contents string) (string, error) {
...
+	return sh(format, true, args)
+}
+
+// ShellMuteOutput run command on shell and get back output and error if get one
+// without logging the output
+func ShellMuteOutput(format string, args ...interface{}) (string, error) {
+	return sh(format, false, args)
+}
+
+func sh(format string, logOutput bool, args ...interface{}) (string, error) {
```
Comment (yutongz) [inline tests/e2e/util/commonUtils.go]: From: [log](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/673/e2e-suite-rbac-no_auth/396/):  `kubectl create namespace [mixer-test-86d2f7b0834b489caeb6917b89]`

If you print `args` in `Shell(...)`, you get `[demo-test-cbd6eaaacfb5427ba65160b3999]`

and then you print `args` in `sh(...)`, you get `[[demo-test-cbd6eaaacfb5427ba65160b3999]]`

Wrapped twice.

File: tests/e2e/framework/framework.go [hunk 2]
```diff
@@ -182,7 +182,7 @@ func (c *CommonConfig) saveLogs(r int) error {
 		glog.Errorf("Could not create status file. Error %s", err)
 		return err
 	}
-	if r != 0 && *logProvider == "stackdriver" { // fetch logs only if tests failed
+	if *logProvider == "stackdriver" { // fetch logs only if tests failed
```
Comment (sebastienvas) [inline tests/e2e/framework/framework.go]: Please eemove logProvider here as well as calls from the shell script.

Comment (chxchx) [reply tests/e2e/framework/framework.go]: Done

File: tests/e2e/framework/testInfo.go [hunk 3]
```diff
@@ -163,96 +130,54 @@ func (t testInfo) Update(r int) error {
...
 			return err
 		}
-		defer func() {
-			if err := f.Close(); err != nil {
-				glog.Warningf("Error during closing file: %v\n", err)
+		containers := strings.Split(containersString, " ")
+		for _, container := range containers {
+			path := filepath.Join(t.LogsPath, fmt.Sprintf("%s_container:%s.log", pod, container))
+			f, err := os.Create(path)
+			if err != nil {
```
Comment (sebastienvas) [inline tests/e2e/framework/testInfo.go]: Add a defer close here.

Comment (chxchx) [reply tests/e2e/framework/testInfo.go]: Done

**Comment by ldemailly** [comment]
let's maybe not merge this while we have the current issues - unless 300% sure it will only be better and doesn't have its own bugs and has been tried on some test cluster/run

File: tests/e2e/framework/testInfo.go [hunk 4]
```diff
@@ -27,62 +27,29 @@ import (
 	"path/filepath"
 	"regexp"
 	"strings"
```
Comment (sebastienvas) [inline tests/e2e/framework/testInfo.go]: Did you run gazelle, it seems that some build file might need to be updated.

**Comment by sebastienvas** [comment]
@chxchx Could you resolve conflicts. We might need to add your change before @andraxylia 's https://github.com/istio/istio/pull/902

**Comment by andraxylia** [comment]
I had a quick look at this PR and other than conflicts in the shell script arguments, it should not interfere with my PR.

**Comment by sebastienvas** [comment]
W0925 20:51:27.701] panic: runtime error: slice bounds out of range
W0925 20:51:27.701] 
W0925 20:51:27.701] goroutine 1 [running]:
W0925 20:51:27.702] istio.io/istio/tests/e2e/framework.testInfo.FetchAndSaveClusterLogs(0xc420170690, 0x25, 0x97f758, 0x9, 0x0, 0x0, 0xc420170690, 0x25, 0xc42015a420, 0x56, ...)
W0925 20:51:27.702] 	tests/e2e/framework/testInfo.go:171 +0x3aa
W0925 20:51:27.702] istio.io/istio/tests/e2e/framework.(*CommonConfig).saveLogs(0xc420199300, 0x0, 0x1, 0xc420466680)
W0925 20:51:27.702] 	tests/e2e/framework/framework.go:182 +0x231
W0925 20:51:27.702] istio.io/istio/tests/e2e/framework.(*CommonConfig).RunTest(0xc420199300, 0xbccaa0, 0xc42015a300, 0x1b)
W0925 20:51:27.702] 	tests/e2e/framework/framework.go:200 +0xc1
W0925 20:51:27.702] istio.io/istio/tests/e2e/tests/bookinfo/go_default_test.TestMain(0xc42015a300)
W0925 20:51:27.702] 	tests/e2e/tests/bookinfo/demo_test.go:426 +0xdf

**Comment by chxchx** [comment]
@andraxylia PTAL

File: tests/e2e/framework/testInfo.go [hunk 5]
```diff
@@ -163,96 +130,59 @@ func (t testInfo) Update(r int) error {
...
-				if len(log) == 0 || log[len(log)-1] != '\n' {
-					if _, err = f.WriteString("\n"); err != nil {
-						return err
-					}
+			defer func() {
+				if err = f.Close(); err != nil {
+					glog.Warningf("Error during closing file: %v\n", err)
 				}
+			}()
+			dump, err := util.ShellMuteOutput(fmt.Sprintf("kubectl logs %s -n %s", pod, namespace))
```
Comment (ldemailly) [inline tests/e2e/framework/testInfo.go]: this needs %s , containername
#970

**Pull request merged**